scheduler/core/eventsqueue/events.py

```diff
@@ -0,0 +1,68 @@
+from dataclasses import dataclass
+from datetime import datetime, timedelta
+from typing import List
+
+from lucupy.minimodel import Resource, Conditions
+
+from scheduler.core.meta import AbstractDataclass, FrozenAbstractDataclass
+
+
+@dataclass(frozen=True)
+class Interruption(FrozenAbstractDataclass):
+    """
+    Parent class for any interruption in the night that would
+    cause a new schedule to be created.
+    """
+    start: datetime
+    reason: str
+
+    def __str__(self):
+        return self.__class__.__name__
+
+
+@dataclass
+class Blockage(AbstractDataclass):
+    """
+    Parent class for any event in the night that would block
+    time slots though the night.
+    """
+    start: datetime
+    reason: str
+    end: datetime = None  # needs a resume event
+
+    def ends(self, end: datetime) -> None:
+        self.end = end
+
+    def time_loss(self) -> timedelta:
+        if self.end:
```

Comment:
Maybe this should be to the end of the night if the Blockage is not resolved? Again, just thinking out loud... the fact that these are still questions that we haven't resolved or discussed makes me think we might need a clearer direction moving forward.

Reply:
That's probably true, I did not add to much logic on it because we were focusing on weather changes. I think for Validation/Simulation we are going to know the end event. The problem is if the event last for more than one night. I agree we should be clearer on that.

Reply:
Let's not worry too much about this now, and focus on weather changes specifically. Then once we make sure that those are working, we can decide how we want to model other types of events.